src/core/currency/wallet/currency-wallet-api.ts

```diff
@@ -110,6 +110,12 @@ export function makeCurrencyWalletApi(
       otherMethods[name] = (...args) => method(walletInfo.keys, ...args)
     }
   }
+  if (plugin.currencyInfo.chainDisplayName == null) {
+    plugin.currencyInfo.chainDisplayName = plugin.currencyInfo.displayName
+  }
+  if (plugin.currencyInfo.assetDisplayName == null) {
+    plugin.currencyInfo.assetDisplayName = plugin.currencyInfo.displayName
+  }
```

Comment:
This upgrade logic won't run until we have a wallet, which is too late, because the `currencyInfo` is also available on the `EdgeCurrencyConfig` object as well.

To do the upgrade as early as possible, please more this logic to the `CORE_PLUGINS_ADDED` redux action in plugins-reducer.ts, which is where the core first takes the wrapping paper off the plugins.